File: GameNight.Api/DTOs/Events/EventDtos.cs
namespace GameNight.Api.DTOs.Events;

public record EventDto(
    Guid Id,
    Guid HostUserId,
    string Title,
    string? Description,
    string? GameTitle,
    string? GameCategory,
    string EventDate,
    string StartTime,
    int DurationMinutes,
    int SetupMinutes,
    string? AddressLine1,
    string? City,
    string? State,
    string? PostalCode,
    string? LocationDetails,
    string? DifficultyLevel,
    int MaxPlayers,
    int ConfirmedCount,
    bool IsPublic,
    bool IsCharityEvent,
    string Status,
    UserSummaryDto? Host,
    List<RegistrationDto>? Registrations,
    List<EventItemDto>? Items,
    DateTime CreatedAt
);

public record EventSummaryDto(
    Guid Id,
    string Title,
    string? GameTitle,
    string? GameCategory,
    string EventDate,
    string StartTime,
    int DurationMinutes,
    string? City,
    string? State,
    string? DifficultyLevel,
    int MaxPlayers,
    int ConfirmedCount,
    bool IsPublic,
    bool IsCharityEvent,
    string Status,
    UserSummaryDto? Host
);

public record CreateEventDto(
    string Title,
    string? Description,
    string? GameTitle,
    string? GameCategory = null,
    string EventDate = "",
    string StartTime = "",
    int DurationMinutes = 120,
    int SetupMinutes = 15,
    string? AddressLine1 = null,
    string? City = null,
    string? State = null,
    string? PostalCode = null,
    string? LocationDetails = null,
    string? DifficultyLevel = null,
    int MaxPlayers = 4,
    bool IsPublic = true,
    bool IsCharityEvent = false,
    string Status = "draft"
);

public record UpdateEventDto(
    string Title,
    string? Description,
    string? GameTitle,
    string? GameCategory,
    string EventDate,
    string StartTime,
    int DurationMinutes,
    int SetupMinutes,
    string? AddressLine1,
    string? City,
    string? State,
    string? PostalCode,
    string? LocationDetails,
    string? DifficultyLevel,
    int MaxPlayers,
    bool IsPublic,
    bool IsCharityEvent,
    string Status
);

public record RegistrationDto(
    Guid Id,
    Guid UserId,
    string Status,
    UserSummaryDto? User,
    DateTime RegisteredAt
);

public record EventItemDto(
    Guid Id,
    string ItemName,
    string ItemCategory,
    int QuantityNeeded,
    Guid? ClaimedByUserId,
    string? ClaimedByName,
    DateTime? ClaimedAt
);

public record CreateEventItemDto(
    string ItemName,
    string ItemCategory = "other",
    int QuantityNeeded = 1
);

public record UserSummaryDto(
    Guid Id,
    string? DisplayName,
    string? AvatarUrl
);

File: android/app/src/main/java/com/sasquatsh/app/data/remote/dto/EventDtos.kt
package com.sasquatsh.app.data.remote.dto

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class UserSummaryDto(
    @Json(name = "id") val id: String,
    @Json(name = "displayName") val displayName: String?,
    @Json(name = "avatarUrl") val avatarUrl: String?,
    @Json(name = "isFoundingMember") val isFoundingMember: Boolean?,
    @Json(name = "isAdmin") val isAdmin: Boolean?,
)

@JsonClass(generateAdapter = true)
data class EventSummaryDto(
    @Json(name = "id") val id: String,
    @Json(name = "title") val title: String,
    @Json(name = "gameTitle") val gameTitle: String?,
    @Json(name = "gameCategory") val gameCategory: String?,
    @Json(name = "gameSystem") val gameSystem: String?,
    @Json(name = "eventDate") val eventDate: String,
    @Json(name = "startTime") val startTime: String?,
    @Json(name = "timezone") val timezone: String?,
    @Json(name = "durationMinutes") val durationMinutes: Int?,
    @Json(name = "city") val city: String?,
    @Json(name = "state") val state: String?,
    @Json(name = "difficultyLevel") val difficultyLevel: String?,
    @Json(name = "maxPlayers") val maxPlayers: Int,
    @Json(name = "hostIsPlaying") val hostIsPlaying: Boolean?,
    @Json(name = "confirmedCount") val confirmedCount: Int?,
    @Json(name = "isPublic") val isPublic: Boolean?,
    @Json(name = "isCharityEvent") val isCharityEvent: Boolean?,
    @Json(name = "minAge") val minAge: Int?,
    @Json(name = "status") val status: String?,
    @Json(name = "primaryGameThumbnail") val primaryGameThumbnail: String?,
    @Json(name = "host") val host: UserSummaryDto?,
)

@JsonClass(generateAdapter = true)
data class EventDetailDto(
    @Json(name = "id") val id: String,
    @Json(name = "title") val title: String,
    @Json(name = "description") val description: String?,
    @Json(name = "gameTitle") val gameTitle: String?,
    @Json(name = "gameCategory") val gameCategory: String?,
    @Json(name = "gameSystem") val gameSystem: String?,
    @Json(name = "eventDate") val eventDate: String,
    @Json(name = "startTime") val startTime: String?,
    @Json(name = "timezone") val timezone: String?,
    @Json(name = "durationMinutes") val durationMinutes: Int?,
    @Json(name = "city") val city: String?,
    @Json(name = "state") val state: String?,
    @Json(name = "postalCode") val postalCode: String?,
    @Json(name = "address") val address: String?,
    @Json(name = "difficultyLevel") val difficultyLevel: String?,
    @Json(name = "maxPlayers") val maxPlayers: Int,
    @Json(name = "hostIsPlaying") val hostIsPlaying: Boolean?,
    @Json(name = "isPublic") val isPublic: Boolean?,
    @Json(name = "isCharityEvent") val isCharityEvent: Boolean?,
    @Json(name = "minAge") val minAge: Int?,
    @Json(name = "status") val status: String?,
    @Json(name = "groupId") val groupId: String?,
    @Json(name = "isMultiTable") val isMultiTable: Boolean?,
    @Json(name = "host") val host: UserSummaryDto?,
    @Json(name = "registrations") val registrations: List<RegistrationDto>?,
    @Json(name = "items") val items: List<EventItemDto>?,
    @Json(name = "games") val games: List<EventGameDto>?,
)

@JsonClass(generateAdapter = true)
data class RegistrationDto(
    @Json(name = "id") val id: String,
    @Json(name = "userId") val userId: String,
    @Json(name = "status") val status: String,
    @Json(name = "registeredAt") val registeredAt: String?,
    @Json(name = "user") val user: UserSummaryDto?,
)

@JsonClass(generateAdapter = true)
data class EventItemDto(
    @Json(name = "id") val id: String,
    @Json(name = "itemName") val itemName: String,
    @Json(name = "itemCategory") val itemCategory: String?,
    @Json(name = "quantityNeeded") val quantityNeeded: Int?,
    @Json(name = "claimedByUserId") val claimedByUserId: String?,
    @Json(name = "claimedAt") val claimedAt: String?,
    @Json(name = "claimedBy") val claimedBy: UserSummaryDto?,
)

@JsonClass(generateAdapter = true)
data class EventGameDto(
    @Json(name = "id") val id: String,
    @Json(name = "bggId") val bggId: Int?,
    @Json(name = "gameName") val gameName: String,
    @Json(name = "thumbnailUrl") val thumbnailUrl: String?,
    @Json(name = "minPlayers") val minPlayers: Int?,
    @Json(name = "maxPlayers") val maxPlayers: Int?,
    @Json(name = "playingTime") val playingTime: Int?,
    @Json(name = "isPrimary") val isPrimary: Boolean?,
    @Json(name = "isAlternative") val isAlternative: Boolean?,
)
